mlflow/models/evaluation/default_evaluator.py

```diff
@@ -614,7 +662,53 @@ def plot_pr_curve():
 
         self._log_pandas_df_artifact(per_class_metrics_collection_df, "per_class_metrics")
 
-    def _evaluate_custom_metrics(self):
+    def _log_custom_metric_artifact(
+        self, artifact_name, raw_artifact, custom_metric_index, custom_metric_name
+    ):
+        if isinstance(raw_artifact, EvaluationArtifact):
+            return raw_artifact
+
+        exception_header = _get_custom_metric_exception_header(
+            custom_metric_index, custom_metric_name
+        )
+
+        inferred_from_path, inferred_type, inferred_ext = _infer_artifact_type_and_ext(
+            artifact_name, raw_artifact, exception_header
+        )
+        artifact_file_name = _gen_log_key(artifact_name, self.dataset_name) + inferred_ext
+        artifact_file_local_path = self.temp_dir.path(artifact_file_name)
+
+        if inferred_from_path:
+            shutil.copyfile(raw_artifact, artifact_file_local_path)
+        elif inferred_type is CsvEvaluationArtifact:
+            raw_artifact.to_csv(artifact_file_local_path, index=False)
+        elif inferred_type is NumpyEvaluationArtifact:
+            np.save(artifact_file_local_path, raw_artifact, allow_pickle=False)
+        elif inferred_type is ImageEvaluationArtifact:
+            raw_artifact.savefig(artifact_file_local_path)
```

Comment:
Are we missing cases for `ParquetEvaluationArtifact` and `JsonArtifact` here? If not, can we add an inline comment explaining why (also, if these classes aren't necessary, can we delete them?)?

Are we missing corresponding test coverage?

Reply:
`ParquetEvaluationArtifact` isn't here because it can only be supplied through file paths. Specifically, if user supplies a `DataFrame` object, we default to saving it as a `CsvEvaluationArtifact`. `JsonEvaluationArtifact` isn't here because    the last `else` clause handles that case. And since `_infer_artifact_type_and_ext` defaults to `JsonEvaluationArtifact` if it can't match other types, this case is handled. I'll explain this in the code. We're not missing test coverage, if you refer to the test cases for both `_infer_artifact_type_and_ext` and `evaluate` with custom metrics, both have cases that tests for parquet and json artifacts.

Reply:
I see. In that case, `ParquetEvaluationArtifact` and `JsonEvaluationArtifact` aren't used anywhere. Can we remove them?

Reply:
Discussed offline, these are returned to users.